src/main/java/duke/core/Duke.java

```diff
@@ -0,0 +1,128 @@
+            this.tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Main loop for processing user commands and executing corresponding actions.
+     */
+    public void echo() {
+        while (ui.hasNext()) {
+            String msg = ui.getInput();
+            Parser parser = new Parser(msg);
+
+            try {
+                if (msg.toUpperCase().startsWith(Command.BYE.name()) && parser.parseBye()) {
+                    ui.formatReply(ui.exit());
+                    break;
+                } else if (msg.toUpperCase().startsWith(Command.LIST.name()) && parser.parseList()) {
+                    ui.formatReply(ui.showAllTasks(tasks.getTasks()));
+                } else if (msg.toUpperCase().startsWith(Command.FIND.name())) {
+                    String description = parser.parseFind();
+                    ui.formatReply(ui.showFoundTasks(tasks.findTasks(description)));
+                } else if (msg.toUpperCase().startsWith(Command.MARK.name())) {
+                    int taskIndex = parser.parseMark();
+                    Task t = tasks.markTask(taskIndex);
+                    ui.formatReply(ui.showMarkTaskMessage(t));
+                    tasks.saveToStorage(storage);
+                } else if (msg.toUpperCase().startsWith(Command.UNMARK.name())) {
+                    int taskIndex = parser.parseUnmark();
+                    Task t = tasks.unmarkTask(taskIndex);
+                    ui.formatReply(ui.showUnmarkTaskMessage(t));
+                    tasks.saveToStorage(storage);
+                } else if (msg.toUpperCase().startsWith(Command.DELETE.name())) {
+                    int taskIndex = parser.parseDelete();
+                    Task t = tasks.deleteTask(taskIndex);
+                    ui.formatReply(ui.showDeleteTaskMessage(t, tasks.getTasks().size()));
+                    tasks.saveToStorage(storage);
+                } else if (msg.toUpperCase().startsWith(Duke.Command.TODO.name())) {
+                    Task t = parser.parseTodo();
+                    tasks.addTask(t);
+                    ui.formatReply(ui.showAddTaskMessage(t, tasks.getTasks().size()));
+                    tasks.saveToStorage(storage);
+                } else if (msg.toUpperCase().startsWith(Command.DEADLINE.name())) {
+                    Task t = parser.parseDeadline();
+                    tasks.addTask(t);
+                    ui.formatReply(ui.showAddTaskMessage(t, tasks.getTasks().size()));
+                    tasks.saveToStorage(storage);
+                } else if (msg.toUpperCase().startsWith(Command.EVENT.name())) {
+                    Task t = parser.parseEvent();
+                    tasks.addTask(t);
+                    ui.formatReply(ui.showAddTaskMessage(t, tasks.getTasks().size()));
+                    tasks.saveToStorage(storage);
+                } else {
+                    ui.displayError("Unable to understand the command. Please enter a valid command.");
+                }
```

Comment:
Any particular reason you use if-else statements instead of using a switch statement?